### PR TITLE
A better approach to headers

### PR DESCRIPTION
**Pull request by drusellers**
- thank you npgsql

So, it occurred to me that NPGSQL might just do this for me, and sure enough it did.

### CODE REVIEW

File: hucache.http/caches/Postgres.fs [hunk 1]
```diff
@@ -20,36 +20,28 @@ let connectionString =
...
+
+let deserialize (raw : IDictionary<string, string>) : Map<string, string> =
     match raw with 
     | null -> Map.empty
-    | r -> r.Trim().Split [|','|]
-                |> Seq.map (fun h -> Regex.Split(h, "=>"))
-                |> Seq.map (fun a -> ((s a 0), (s a 1)))
-                |> Map.ofSeq
+    | r -> (r :> seq<_>)
+           |> Seq.map (|KeyValue|)
```
Comment (dahlbyk) [inline hucache.http/caches/Postgres.fs:35]: Didn't know this active pattern was built in... :cool:

File: hucache.http/Cache.fs [hunk 2]
```diff
@@ -5,10 +5,10 @@ open hucache.http.types
...
 let lookup (key: IssueKey) : FullPayload option =
-    match getFromDb (key) with
+    match loadIssue key with
     | Some payload -> Some(payload)
     | None -> 
         let issue = fetchIssue key
         match issue with 
-        | Some i -> Some((storeInDb (key, i)))
+        | Some i -> Some((storeIssue (key, i)))
         | None -> None
```
Comment (dahlbyk) [inline hucache.http/Cache.fs:14]: I will merge as-is, but you might look into Option combinators to make this a little simpler.

Comment (drusellers) [inline hucache.http/Cache.fs:14]: what are option combinators?

Comment (dahlbyk) [inline hucache.http/Cache.fs:14]: Check out `bind` and `map`: 
https://msdn.microsoft.com/visualfsharpdocs/conceptual/core.option-module-%5bfsharp%5d

Comment (drusellers) [inline hucache.http/Cache.fs:14]: Not sure how `| Some i -> map storeInDb (key, i)` is better than `Some` - therefor I'm missing your lesson. :(

Comment (dahlbyk) [inline hucache.http/Cache.fs:14]: No specific lesson, just making sure you're aware. If `storeIssue` args were curried you could do something like:

``` fsharp
let lookup (key: IssueKey) : FullPayload option =
    match loadIssue key with
    | Some payload -> Some(payload)
    | None -> fetchIssue key |> Option.map (storeIssue key)
```

**Pull request merged**